src/System.DirectoryServices.AccountManagement/src/System/DirectoryServices/AccountManagement/Utils.cs

```diff
@@ -969,7 +969,7 @@ internal static bool IsMachineDC(String computerName)
             try
             {
                 if (null == computerName)
-                    err = UnsafeNativeMethods.DsRoleGetPrimaryDomainInformation(IntPtr.Zero, UnsafeNativeMethods.DSROLE_PRIMARY_DOMAIN_INFO_LEVEL.DsRolePrimaryDomainInfoBasic, out dsRoleInfoPtr);
```

Comment:
The compiler will cast 'null' to System.String no problem, but throws an error trying to do the same (explicitly or implicitly) for IntPtr.Zero.   Not sure how\why the code compiled ok in its original home, but switching to 'null' solved the problem just fine and is also more consistent with the if clause immediately above this line..